tests/test_pipeline.py

```diff
@@ -0,0 +1,235 @@
+    with pytest.raises(ValidationError):
+        ta_gt.validate_python(0)
+
+    ta_gt_strict = TypeAdapter(Annotated[int, validate_as(int, strict=True).gt(0)])
+    assert ta_gt_strict.validate_python(1) == 1
+    with pytest.raises(ValidationError):
+        ta_gt_strict.validate_python('1')
+    with pytest.raises(ValidationError):
+        ta_gt_strict.validate_python(0)
+
+
+def test_parse_str_to_int() -> None:
+    ta = TypeAdapter(Annotated[int, validate_as(str).str_strip().validate_as(int)])
+    assert ta.validate_python('1') == 1
+    assert ta.validate_python(' 1 ') == 1
+    with pytest.raises(ValidationError):
+        ta.validate_python('a')
+
+
+def test_predicates() -> None:
+    ta_int = TypeAdapter(Annotated[int, validate_as(int).predicate(lambda x: x % 2 == 0)])
+    assert ta_int.validate_python(2) == 2
+    with pytest.raises(ValidationError):
+        ta_int.validate_python(1)
+
+    ta_str = TypeAdapter(Annotated[str, validate_as(str).predicate(lambda x: x != 'potato')])
+    assert ta_str.validate_python('tomato') == 'tomato'
+    with pytest.raises(ValidationError):
+        ta_str.validate_python('potato')
+
+
+@pytest.mark.parametrize(
+    'model, expected_val_schema, expected_ser_schema',
+    [
+        (
+            Annotated[Union[int, str], validate_as() | validate_as(str)],
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+        ),
+        (
+            Annotated[int, validate_as() | validate_as(str).validate_as(int)],
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+            {'type': 'integer'},
+        ),
+        (
+            Annotated[int, validate_as() | validate_as(str).transform(int)],
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+        ),
+        (
+            Annotated[int, validate_as() | validate_as(str).transform(int).validate_as(int)],
+            {'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+            {'type': 'integer'},
+        ),
+        (
+            Annotated[int, validate_as(int).gt(0).lt(100)],
+            {'type': 'integer', 'exclusiveMinimum': 0, 'exclusiveMaximum': 100},
+            {'type': 'integer', 'exclusiveMinimum': 0, 'exclusiveMaximum': 100},
+        ),
+        (
+            Annotated[int, validate_as(int).gt(0) | validate_as(int).lt(100)],
+            {'anyOf': [{'type': 'integer', 'exclusiveMinimum': 0}, {'type': 'integer', 'exclusiveMaximum': 100}]},
+            {'anyOf': [{'type': 'integer', 'exclusiveMinimum': 0}, {'type': 'integer', 'exclusiveMaximum': 100}]},
+        ),
+        (
+            Annotated[List[int], validate_as().len(0, 100)],
+            {'type': 'array', 'items': {'type': 'integer'}, 'maxItems': 100},
+            {'type': 'array', 'items': {'type': 'integer'}, 'maxItems': 100},
+        ),
```

Comment:
```suggestion
        ),
        (
            Annotated[int, validate_as(str).transform(int)],
            {'type': 'string'},
            {'type': 'integer'},
        ),
```
Does this work? If so I think it's worth including as a test case here, since the others all seem to always include the output type as one of the allowed validation inputs; I think it's worth being explicit that if you don't make your input idempotent, it won't be.

Reply:
Agreed

Reply:
@adriangb this test is failing with this addition

Reply:
The ser schema isn't right